salt/modules/linux_lvm.py

```diff
@@ -486,10 +486,21 @@ def lvresize(size, lvpath):
 
 
         salt '*' lvm.lvresize +12M /dev/mapper/vg1-test
+        salt '*' lvm.lvresize lvpath=/dev/mapper/vg1-test extents=+100%FREE
+
     '''
-    ret = {}
-    cmd = ['lvresize', '-L', six.text_type(size), lvpath]
-    cmd_ret = __salt__['cmd.run_all'](cmd, python_shell=False)
-    if cmd_ret['retcode'] != 0:
-        return {}
-    return ret
+    if size and extents:
+        return 'Error: Please specify only one of size or extents'
```

Comment:
This is changing the expected return structure it seems. Is that intended? Is that handled elsewhere?

Reply:
Yes, but this will be returned iff _extents_ and size will be used. And since _extents_ is introduced here, it won't affect others.

Reply:
Yeah, sure. Mostly I am wondering if we can return a more consistent data structure and _log_ the error message. How would a user know when this function is successful or not without examining the string? If this function is going to be changed, it should be consistent with the other functions in this execution module. I don't think we should change the return data structure when people are expecting something else unless there is a good reason.

Reply:
So the retrun 'Error...' is the same style as the rest of the functhions in this module.
The return value type in line 506 is a dict, because this fucntion returned an (empty) dict consistenly.

Do you want me to change the error strings to dict? or the dict (line 506) to string?

Reply:
I think we should keep the return as a dict and log the error. Something like this:
```
if size and extents:
    log.error('Error: Please specify only one of size or extents')
    return {}
```